File: web/js/utils/findcontrolbyuuid.js


export function findControlByUuid(uuid, controls) {
    if (!Array.isArray(controls)) return null;
    
    for (const c of controls) {
        if (c.uuid === uuid) {
            return c;
        }

        
        if (Array.isArray(c.controls)) {
            const found = findControlByUuid(uuid, c.controls);
            if (found !== null) {
                return found;
            }
        }
    }

    return null;
}

File: kotway/utils/find_control_by_uuid.py
from ..core.controls.parentcontrol import ParentControl

def find_control_by_uuid (uuid:str, controls: list):
    for c in controls:
        if uuid == c.uuid:
            return c

        if isinstance(c, ParentControl):
            cc = find_control_by_uuid(uuid, c.controls)
            if cc != None:
                return cc